src/mrnet/core/reactions.py

```diff
@@ -1860,6 +1975,134 @@ def from_dict(cls, d):
         return reaction
 
 
+def general_graph_rep(reaction: Reaction) -> nx.DiGraph:
+    """
+    A method to convert a reaction type object into a general graph representation.
+    Can handle reactions with arbitrary numbers of reactants and products
+    Args:
+       :param reaction:(any of the reaction class object, ex. RedoxReaction,
+       IntramolSingleBondChangeReaction)
+    """
```

Comment:
Took me a minute to grok what was going on, but this is a really elegant solution! And soon, it'll be even shorter!